### ISSUE TITLE
[/kit][Config] VitePlugin options typed in Config

### ISSUE THREAD

**Issue by WillsterJohnson** [sveltejs/kit#9935]
### Describe the problem

**TL:DR;**

the `vitePlugin` key in `@sveltejs/kit.Config` is a bit difficult to discover if you don't already know to look for it. Adding it as a typed option would help with this.
Also I'm happy to implement & PR this if the team approve of the idea.

**Full detail**

I just spent the last half an hour trying to figure out how to enable the inspector.
I couldn't recall it's name, it wasn't in the docs on https://svelte.dev/ or https://kit.svelte.dev/, all I could remember about it is that Rich demoed it in one of their talks sometime last year, and that it was enabled by a config option *somewhere*.

I managed to find an issue from it's experimental days which gave me the name, but the config was in `vite-plugin-svelte`, no use for my SvelteKit project. A few dozen permutations of `svelte vite inspector sveltekit` in google and I managed to find #7390, and then later #5888 which gave me a clearer example of how to enable the inspector.

This feature isn't too hard to find once you know what to search for, but stumbling upon it by accident is probably never going to happen, meaning if you don't know, you *won't* know.

### Describe the proposed solution

Following on from #5888

Instead of having to use this JSDoc in `svelte.config.js`; 
```js
/** @type {import('@sveltejs/kit').Config & { vitePlugin?: import("@sveltejs/vite-plugin-svelte").PluginOptions}} */
```
then manually add `@sveltejs/vite-plugin-svelte` to `devDependencies` in order for TypeScript to bother finding it (pnpm &/ monorepo only?) we could simply use the default JSDoc and have the `vitePlugin` key be typed internally, much like how `compilerOptions` is typed even though it refers to something other than `@sveltejs/kit`.

I don't see it taking more than;
```diff
+import { PluginOptions } from "@sveltejs/vite-plugin-svelte";
...
export interface Config {
...
+  vitePlugin?: PluginOptions;
   /** Any additional options required by tooling that integrates with Svelte. */
   [key: string]: any;
}
```
(in terms of the `.d.ts` output, internally I assume it would be somewhat different to the above)

This way `ctrl+space` in VSCode will show this as one of six (four if you've used `create-svelte`) possible keys at the base of the config object, making discovering the inspector & other `vite-plugin-svelte` features significantly easier.

### Alternatives considered

**Changing the default JSDoc in `svelte.config.js` for `create-svelte`**

The main problem with the JSDoc to type the `vitePlugin` key is that it's pretty ugly, expecially for a config file. And if by default we're suggesting that the `vitePlugin` key should have a specific type, then... give it the specific type.
This also wouldn't effect existing projects who are updating their dependencies, meaning any project created before this change would never know until a developer begins work on a new project (or reads this in the patch notes).

**Do nothing**

Sure, but we're left with the issue; `vite-plugin-svelte`'s features and functionality are near impossible to discover for users who quite reasonably assume `create-svelte` and the Svelte/SvelteKit docs will give them everything they need to build.
I try to keep up with the ecosystem, but it still took me half an hour to figure this out. I don't think that was entirely my own fault.

### Importance

would make my life easier

### Additional Information

```diff
Importance:
-would make my life easier
+would've made my life easier
```
This change probably won't benefit me, but it could benefit others like me, as well as everyone who knows less about `vite-plugin-svelte` than I do (it's not a high bar, but it's a bar) who would find it's configurable features useful.

**Is this a breaking change?**

If someone out there is using `vitePlugin` as the config key for their interoperating tools, then they may have to update their types, or even switch over to a different key entirely.
However, they are already incompatible with the base SvelteKit library, and conflicting namespaces with a core part of the system you're hooking into is a brave idea.

Technically, yeah this could break some stuff. Realistically, there's probably nobody doing that, and if there are it's unfortunately kinda their fault.

---

*I'm happy to get this working & send a PR if the team are happy for this to exist*

**Comment by tcc-sejohnson** [comment]
```diff
Importance:
-would make my life easier
+would've made my life easier
```

lol, love it
